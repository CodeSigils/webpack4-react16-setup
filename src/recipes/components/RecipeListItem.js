import React from 'react';
import PropTypes from 'prop-types';

const RecipeListItem = ({ recipe, onClick, onFavorited, favorited }) => (
  <li
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={() => onClick(recipe.id)}
  >
    <span
      className="mr1 red"
      onClick={e => {
        e.stopPropagation();
        onFavorited(recipe.id);
      }}
      role="img"
      aria-label="favorite"
    >
      {favorited ? 'ðŸ’–' : 'â˜‘'}
    </span>
    <span>{recipe.name}</span>
    <span>{recipe.category}</span>
  </li>
);

RecipeListItem.propTypes = {
  recipe: PropTypes.object,
  onClick: PropTypes.func,
  onFavorited: PropTypes.func,
  favorited: PropTypes.bool,
};

RecipeListItem.defaultProps = {
  recipe: {},
};

export default RecipeListItem;
