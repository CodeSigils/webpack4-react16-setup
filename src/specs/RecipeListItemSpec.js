import React from 'react';
import renderer from 'react-test-renderer';
import RecipeListItem from '../recipes/components/RecipeListItem';

const testRecipe = {
  id: 1,
  name: 'Test recipe',
  category: 'Test recipe category',
};

describe('<RecipeListItem/>', () => {
  // 1. test for recipe null
  test('Should not break if there is no recipe passed', () => {
    // Transform  the component into a JSON structure
    const component = renderer.create(<RecipeListItem />);
    const tree = component.toJSON();
    // Tell Jest to check if tree muches the snapshot
    expect(tree).toMatchSnapshot();
  });

  // 2. test for recipe not null
  test('Should correctly render recipe', () => {
    const component = renderer.create(<RecipeListItem recipe={testRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  // 3. test for favorites bool state
  test('Should render favorite state', () => {
    const component = renderer.create(<RecipeListItem favorited={false} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
