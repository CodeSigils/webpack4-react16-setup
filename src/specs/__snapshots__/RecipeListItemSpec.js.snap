// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem/> Should correctly render recipe 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1 red"
    onClick={[Function]}
    role="img"
  >
    ☑
  </span>
  <span>
    Test recipe
  </span>
  <span>
    Test recipe category
  </span>
</li>
`;

exports[`<RecipeListItem/> Should not break if there is no recipe passed 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1 red"
    onClick={[Function]}
    role="img"
  >
    ☑
  </span>
  <span />
  <span />
</li>
`;

exports[`<RecipeListItem/> Should render favorite state 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mr1 red"
    onClick={[Function]}
    role="img"
  >
    ☑
  </span>
  <span />
  <span />
</li>
`;
